src/main/java/seedu/address/model/UserPrefs.java

```diff
@@ -68,7 +68,7 @@ public boolean equals(Object other) {
         UserPrefs o = (UserPrefs) other;
 
         return guiSettings.equals(o.guiSettings)
-                && addressBookFilePath.equals(o.addressBookFilePath);
+                && Objects.equals(addressBookFilePath.toAbsolutePath(), o.addressBookFilePath.toAbsolutePath());
```

Comment:
I had to change this because of dependency upgrades to `jackson` and how it affects deserialization of file paths. 

I found out that `Path::equals` checks for string equality of paths which is a little weird, since `preference.json` and `~/preference.json` is the same location but it will fail the equality check. Therefore I went with a check to the absolute paths of the `Path` object which imo is better test